x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -272,10 +314,62 @@ static void validate(String jobId, Job job) {
                 maxAvailableCount = availableCount;
                 minLoadedNodeByCount = node;
             }
+
+            String machineMemoryStr = nodeAttributes.get(MachineLearning.MACHINE_MEMORY_NODE_ATTR);
+            long machineMemory = -1;
+            // TODO: remove leniency and reject the node if the attribute is null in 7.0
```

Comment:
I thought about doing this but it's really separate to the change of this PR, and I thought it might create a merge clash in the future.  #29963 is tracking the need to do the 7.0.0 TODOs.  (That issue says by alpha1, which we've missed, but I don't think that's critical.)  But I'll do it as soon as these feature branches are merged - that should be in time for alpha2 as 6.6 FF is before alpha2.

Reply:
👍